### PR TITLE
Main models

### LINKED ISSUE #12: Create main models
- [x] Exam
- [x] Exercise
- [x] Exercise Test Case
- [x] Exercise Solution
- [x] Implement ```equals```, ```hashcode```, ```toString``` methods
- [x] Define models API (i.e what can you do with each of them as, for example, change exam duration, or exercise question).

### Exam
- Starting moment
- Duration
- List of exercises
- Minimum acceptable score **(will be implemented in #14)** 
- Total Score **(will be implemented in #14)** 
- Owner teacher (or **teachers**, in plural)? **(Will be implemented in #18)**

### Exercise
- Exam to which it belongs
- Question
- Granted Score **(will be implemented in #14)** 
- List of public test cases
- List of private test cases

### Test Case
- Exercise to which it belongs
- Inputs
- Expected Outputs
- Timeout (optional?) **(Will be implemented in #19)**
- Score Percentage (or Granted Score) **(will be implemented in #14)** 
- Is private? (should we add this in the test case model?) (Implemented by ```visibility``` field)

### Exercise solution
- Exercise to which it belongs
- Answer
- Owner Student **(Will be implemented in #18)**

### PR DESCRIPTION
**Pull request by juanmbellini**
This pull request creates main models

Closes #12

### CODE REVIEW

File: evaluations-service-core/evaluations-service-models/src/main/java/ar/edu/itba/cep/evaluations_service/models/ExerciseSolutionResult.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    // Helpers
+    // ================================
+
+    /**
+     * An enum indicating whether the result is "approved" or "failed".
+     */
+    public enum Result {
+        APPROVED,
+        FAILED,
+        ;
```
Comment (lobo) [inline evaluations-service-core/evaluations-service-models/src/main/java/ar/edu/itba/cep/evaluations_service/models/ExerciseSolutionResult.java:122]: does this compile? `,;`?

Comment (juanmbellini) [reply evaluations-service-core/evaluations-service-models/src/main/java/ar/edu/itba/cep/evaluations_service/models/ExerciseSolutionResult.java:122]: Yes, enums and arrays can have a ```,``` in the end, and the ```;``` can also be present.

Comment (lobo) [reply evaluations-service-core/evaluations-service-models/src/main/java/ar/edu/itba/cep/evaluations_service/models/ExerciseSolutionResult.java:122]: I'd try to make the `description`, `duration` and `startingAt` compulsory for the creation of the object, somehow. @juanmbellini

Comment (juanmbellini) [reply evaluations-service-core/evaluations-service-models/src/main/java/ar/edu/itba/cep/evaluations_service/models/ExerciseSolutionResult.java:122]: Yes, in another PR

**Pull request merged**